src/Microsoft.AspNet.OData.Shared/Query/ExpandQueryBuilder.cs

```diff
@@ -0,0 +1,110 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ExpandQueryBuilder.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    /// <summary>
+    /// Exposes the ability to generate a $expand query string from a payload object.
+    /// </summary>
+    public class ExpandQueryBuilder : IExpandQueryBuilder
+    {
+        /// <inheritdoc />
+        public virtual string GenerateExpandQueryString(object value, IEdmModel model)
```

Comment:
If they generate a different query string, it will be up to them.
Generally a response will process the query options as they are set in the `HttpRequest`